src/output/plugins/visualization/SoundAnalysis.hxx

```diff
@@ -0,0 +1,317 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright The Music Player Daemon Project
+
+#ifndef SOUND_ANALYSIS_HXX_INCLUDED
+#define SOUND_ANALYSIS_HXX_INCLUDED 1
+
+#include "SoundInfoCache.hxx"
+#include "LowLevelProtocol.hxx"
+
+#include <fftw3.h>
+
+#include <algorithm>
+#include <cstddef>
+#include <memory>
+
+#include <math.h>
+
+struct ConfigBlock;
+
+namespace Visualization {
+
+/**
+ * \brief Convenience class for expressing sound analysis parameters exclusive
+ * of the audio format
+ *
+ *
+ * There are any number of parameters governing our analysis of PCM data. Other
+ * than the `AudioFormat`, they are read from configuration at startup and
+ * constant. Rather than force callers to write methods taking many parameters,
+ * this class colects them all in one place, and enforces some constraints on
+ * their values.
+ *
+ *
+ */
+
+class SoundAnalysisParameters
```

Comment:
Why is this a `class` with private fields and not a `struct`? Your design requires getter methods for something that simple, but what's the point of protecting the fields, for having to add code bloat?